Sources/Amplitude/DefaultTrackingOptions.swift

```diff
@@ -10,54 +10,57 @@ public class DefaultTrackingOptions {
     }
 
     public var sessions: Bool {
-        get {
-            autocaptureOptions.contains(.sessions)
-        }
-        set {
-            if newValue {
-                autocaptureOptions.insert(.sessions)
-            } else {
-                autocaptureOptions.remove(.sessions)
-            }
+        didSet {
+            delegate?.didChangeOptions(options: self)
         }
     }
 
     public var appLifecycles: Bool {
-        get {
-            autocaptureOptions.contains(.appLifecycles)
-        }
-        set {
-            if newValue {
-                autocaptureOptions.insert(.appLifecycles)
-            } else {
-                autocaptureOptions.remove(.appLifecycles)
-            }
+        didSet {
+            delegate?.didChangeOptions(options: self)
         }
     }
 
     public var screenViews: Bool {
-        get {
-            autocaptureOptions.contains(.screenViews)
-        }
-        set {
-            if newValue {
-                autocaptureOptions.insert(.screenViews)
-            } else {
-                autocaptureOptions.remove(.screenViews)
-            }
+        didSet {
+            delegate?.didChangeOptions(options: self)
         }
     }
 
-    var autocaptureOptions: AutocaptureOptions
+    weak var delegate: DefaultTrackingOptionsDelegate?
+
+    var toAutocaptureOptions: AutocaptureOptions {
+        return [
+            sessions ? .sessions : [],
+            appLifecycles ? .appLifecycles : [],
+            screenViews ? .screenViews : []
+        ].reduce(into: []) { $0.formUnion($1) }
+    }
 
     public init(
         sessions: Bool = true,
         appLifecycles: Bool = false,
         screenViews: Bool = false
     ) {
-        self.autocaptureOptions = []
         self.sessions = sessions
         self.appLifecycles = appLifecycles
         self.screenViews = screenViews
     }
+
+    convenience init(with delegate: DefaultTrackingOptionsDelegate) {
```

Comment:
Try to make the names self descriptive where possible - compare something like `DefaultTrackingOptions(with: delegate)` to `DefaultTrackingOptions(delegate: delegate)`.  The shorter naming generally works better when the class or method name is related to the parameter `dataTask(for: request)` or something like `minX(for: rect)`.